app/screens/home/search/search.tsx

```diff
@@ -59,62 +80,117 @@ const SearchScreen = ({teamId}: Props) => {
     const [fileInfos, setFileInfos] = useState<FileInfo[]>(emptyFileResults);
     const [fileChannelIds, setFileChannelIds] = useState<string[]>([]);
 
-    const getSearchParams = useCallback((filterValue?: FileFilter) => {
-        const terms = filterValue ? lastSearchedValue : searchValue;
-        const fileExtensions = filterFileExtensions(filterValue || filter);
-        const extensionTerms = fileExtensions ? ' ' + fileExtensions : '';
-        return {
-            terms: terms + extensionTerms,
-            is_or_search: true,
-        };
-    }, [filter, lastSearchedValue, searchValue]);
-
-    const handleSearch = useCallback((debounce(async () => {
-        // execute the search for the text in the navigation text box
-        // handle recent searches
-        // - add recent if doesn't exist
-        // - updated recent createdAt if exists??
-
-        const searchParams = getSearchParams();
-        if (!searchParams.terms) {
-            handleClearSearch();
-            return;
-        }
-        setLoading(true);
-        setShowResults(true);
-        setFilter(FileFilters.ALL);
-        setLastSearchedValue(searchValue);
-        const [postResults, {files, channels}] = await Promise.all([
-            searchPosts(serverUrl, searchParams),
-            searchFiles(serverUrl, teamId, searchParams),
-        ]);
-
-        setFileInfos(files?.length ? files : emptyFileResults);
-        setPostIds(postResults?.order?.length ? postResults.order : emptyPostResults);
-        setFileChannelIds(channels?.length ? channels : emptyChannelIds);
-        setLoading(false);
-    })), [searchValue]);
+    const handleSearch = useRef<(term: string) => void>();
 
     const onSnap = (offset: number) => {
         scrollRef.current?.scrollToOffset({offset, animated: true});
     };
 
+    const {scrollPaddingTop, scrollRef, scrollValue, onScroll, headerHeight, hideHeader} = useCollapsibleHeader<FlatList>(true, onSnap);
+
+    const onSubmit = useCallback(() => {
+        handleSearch.current?.(searchValue);
+    }, [searchValue]);
+
+    const handleClearSearch = useCallback(() => {
+        setSearchValue('');
+        setLastSearchedValue('');
+        setFilter(FileFilters.ALL);
+    }, []);
+
+    const handleCancelSearch = useCallback(() => {
+        handleClearSearch();
+        setShowResults(false);
+    }, [handleClearSearch, showResults]);
+
+    useEffect(() => {
+        handleSearch.current = async (term: string) => {
```

Comment:
Why are we keeping this API inside a useRef?

Reply:
This API call depends on the teamId, which will change in future PRs. If we had it in any other way, handleRecentChange will change, re-rendering all the recent changes. Not a big deal, since the list will be 15 elements tops, but still, less re-renders.